stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -499,8 +499,12 @@ fileprivate struct _JSONKeyedEncodingContainer<K : CodingKey> : KeyedEncodingCon
     }
 
     public mutating func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: Key) -> KeyedEncodingContainer<NestedKey> {
-        let dictionary = NSMutableDictionary()
-        self.container[_converted(key).stringValue] = dictionary
+        let containerKey = _converted(key).stringValue
+        let existingContainer = self.container[containerKey]
```

Comment:
I feel that on the whole, this might be more idiomatically expressed as something like

```swift
let containerKey = _converted(key).stringValue
let dictionary: NSMutableDictionary
if let existingContainer = self.container[containerKey] {
    precondition(existingContainer is NSMutableDictionary, "Attempt to re-encode into nested KeyedEncodingContainer<\(Key.self)> for key \"\(containerKey)\" is invalid: non-keyed container already encoded for this key")
    dictionary = existingContainer as! NSMutableDictionary
} else {
    dictionary = NSMutableDictionary()
    self.container[containerKey] = dictionary
}
```

Same for plist